### PR TITLE
tests: speedup compaction tests

### PR DESCRIPTION
**Pull request by BusyJay**
On my machine, compaction tests take half time as before.

### CODE REVIEW

File: tests/raftstore/test_compact_log.rs [hunk 1]
```diff
@@ -47,35 +48,59 @@ fn test_compact_log<T: Simulator>(cluster: &mut Cluster<T>) {
...
-        assert!(idx > before_state.get_index());
-        assert!(after_state.get_term() > before_state.get_term());
+        let term = after_state.get_term();
+        if idx == before_state.get_index() || term == before_state.get_term() {
+            return false;
+        }
+        if idx - before_state.get_index() < compact_count {
+            return false;
+        }
+        assert!(term > before_state.get_term());
```
Comment (siddontang) [inline tests/raftstore/test_compact_log.rs]: Can Team be increased here?

**Comment by siddontang** [comment]
PTAL @javaforfun

**Comment by javaforfun** [comment]
LGTM

**Comment by BusyJay** [comment]
PTAL

**Comment by BusyJay** [comment]
@nolouch PTAL

**Pull request merged**